Parse/SMTLIB2.cpp

```diff
@@ -888,87 +719,90 @@ SMTLIB2::DeclaredFunction SMTLIB2::declareFunctionOrPredicate(const vstring& nam
   OperatorType* type;
 
   if (rangeSort == AtomicSort::boolSort()) { // predicate
-    symNum = env.signature->addPredicate(name, argSorts.size(), added);
+    symNum = env.signature->addPredicate(name, argSorts.size()+taArity, added);
 
     sym = env.signature->getPredicate(symNum);
 
-    type = OperatorType::getPredicateType(argSorts.size(),argSorts.begin());
+    type = OperatorType::getPredicateType(argSorts.size(),argSorts.begin(),taArity);
 
     LOG1("declareFunctionOrPredicate-Predicate");
   } else { // proper function
-    if (argSorts.size() > 0) {
-      symNum = env.signature->addFunction(name, argSorts.size(), added);
+    if (argSorts.size() > 0 || taArity > 0) {
+      symNum = env.signature->addFunction(name, argSorts.size()+taArity, added);
     } else {
+      // TODO what about taArity here?
```

Comment:
Right, but I was thinking that maybe constants are handled differently which should be preserved even for `taArity > 0`, so for "constants" modulo type arguments. But as far as I can tell, there's no such special handling anywhere.

Reply:
Yeah, as far as I know (with the possible exception of `Z3Interfacing`!) these "constants" are not treated specially. The whole point of doing the whole TFF1 hack is so that sorts are kinda-terms, so in principle there should be very little special handling.